frontstage/common/authorisation.py

```diff
@@ -1,26 +1,68 @@
+import logging
+
+from datetime import datetime
 from functools import wraps
 from flask import render_template
-from jose.jwt import encode, decode
+from jose import JWTError
+from jose.jwt import decode
+from structlog import wrap_logger
+
+from frontstage import app
+from frontstage.exceptions.exceptions import JWTValidationError
+
+
+logger = wrap_logger(logging.getLogger(__name__))
+
+
+def validate(token):
+    logger.debug('event="Validating token"')
+
+    now = datetime.now().timestamp()
+    expires_at = token.get('expires_at')
+    if expires_at:
+        if now >= expires_at:
+            logger.warning('event="Token has expired", expires_at='.format(token))
+            return False
+    else:
+        logger.warning('event="No expiration date in token"')
+        return False
+
+    # if not set('respondent').intersection(token.get('scope', [])):
+    #     # logger.warning('event="unable to validate scope", token="{}"'.format(token))
+    #     return False
+
+    logger.debug('event="Token is valid"')
+    return True
+
 
 def jwt_authorization(request):
-    """
-    Validate an incoming session and only proceed with a decoded session if the session is valid,
-    otherwise render the not-logged-in page.
-    :param request: The current request object
-    """
 
     jwt_secret = 'vrwgLNWEffe45thh545yuby'
     jwt_algorithm = 'HS256'
 
     def extract_session(original_function):
         @wraps(original_function)
         def extract_session_wrapper(*args, **kwargs):
-            if 'authorization' in request.cookies:
-                session = decode(request.cookies['authorization'], jwt_secret, algorithms=jwt_algorithm)
+            encoded_jwt_token = request.cookies.get('authorization')
+            if encoded_jwt_token:
+                logger.debug('event="Attempting to authorize token", "{}"'.format(encoded_jwt_token))
+                try:
+                    jwt_token = decode(request.cookies.get('authorization'), jwt_secret, algorithms=jwt_algorithm)
+                    logger.debug('event="Token decoded successfully"')
+                except JWTError:
+                    logger.warning('event="Unable to decode token", encrypted_jwt_token="{}"'
+                                   .format(encoded_jwt_token))
+                    raise JWTValidationError
+            else:
+                logger.warning('event="No authorization token provided"')
+                raise JWTValidationError
+
+            if app.config['VALIDATE_JWT']:
```

Comment:
You're right, `VALIDATE_JWT` should either be set by an environment or just implied true